### DISCUSSION TITLE
Template Override Extensions

### DISCUSSION THREAD

**Discussion by venomdev** [joomla/joomla-cms#31627] [2. Cold Features]
I have a designer and users who are familiar with the Twig template format and know little to no PHP so I include a few Twig templates in my extensions. I would like the designer or user to be able to use overrides on files that use an extension other than .php

In the com_templates component could we add another few types of extensions?

the line to change is [com_templates/models/template.php#L787](https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_templates/models/template.php#L787)

I think some other extensions that might be helpful to include are:

- .twig.html
- .html.twig
- .blade.html
- .html.blade

Maybe we could add allowed extensions in the system configuration for com_templates under the Supported File Formats section.

**Comment by dgrammatiko** [comment]
For twig there is a solution from @phproberto https://github.com/phproberto/joomla-twig

**Reply by venomdev** [reply]
I am using this library, and it's pretty handy.

The problem happens when I create a template in the extension view folder, it wont be copied over to the template over ride folder for editing. The only files currently copied over are .php files, so I've had to create a load template function that looks for .twig.php files instead of the default .html.twig.

**Reply by dgrammatiko** [reply]
You could do a PR adding `['.html', '.php', '.twig', '.blade']` there. I guess nobody will object that change

**Reply by venomdev** [reply]
I will when I get a chance. At the moment I'm right in the middle of building the extension that needs this feature.
